features/step_definitions/registration_steps.rb

```diff
@@ -1,16 +1,33 @@
 When("I register with a username") do
   visit register_path
   fill_in 'username', with: "Bono"
-  click_button("Create User")
+  click_button("Register")
 end
 
 Then("I have an account") do
-  expect(page).to have_content "Collaboration Contracts"
+  expect(User.find_by(username: "Bono")).to be_truthy
 end
 
 When("I register with a username and with a password") do
   visit register_path
   fill_in 'username', with: "Bono"
   fill_in 'password_digest', with: "password"
-  click_button("Create User")
+  click_button("Register")
+end
+
+When("I register without a username and with a password") do
+  visit register_path
+  fill_in 'password_digest', with: "password"
+  click_button("Register")
+end
+
+Then("I view the Registration form") do
+  expect(User.find_by(username: "Bono")).to be_falsey
```

Comment:
This is interesting. I think it might be fair to say that the requirements are not explicit enough. I expect it is not always true that "I view the Registration form" indicates we are not registered....

I think the feature should be re-worded.

Reply:
Okay. Maybe a "Then I view the registration form" and a "And a user was not created" or something? Just spitballing. @DocOnDev @SeanMNorton